HIP/hip-729.md

```diff
@@ -0,0 +1,117 @@
+---
+hip: 729
+title: Contract Accounts Nonces Externalization
+author: Miroslav Gatsanoga <miroslav.gatsanoga@limechain.tech>
+working-group: Stoyan Panayotov <stoyan.panayotov@limechain.tech>, Ivan Kavaldzhiev <ivan.kavaldzhiev@limechain.tech>, Richard Bair <@rbair23>, Nana Essilfie-Conduah <@Nana-EC>
+type: Standards Track
+category: Service
+needs-council-approval: Yes
+status: Draft
+created: 2023-05-05
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/730
+updated: 2023-05-09
+---
+
+## Abstract
+
+The Hedera Smart Contract Service (HSCS) supports the deployment of contracts. A deployed contract should have a nonce value that depends on the number of other contracts that were created during its own creation.
```

Comment:
nit: something like this maybe
```suggestion
The Hedera Smart Contract Service (HSCS) supports the deployment of contracts. A deployed contract `A` should have a nonce value that reflects the number of other contracts that were created since `A`'s creation.
```

Reply:
Updated.